corehq/apps/data_dictionary/models.py

```diff
@@ -150,6 +169,12 @@ def group_name(self):
             return self.group_obj.name
         return self.group
 
+    def clean(self):
+        if self.description and len(self.description) > 255:
+            raise ValidationError(
+                gettext_lazy("{} property's description should be less 255 characters".format(self.name))
+            )
```

Comment:
TextField field does not provide a `max_length` attribute to specify a limit. This validation was needed to be done explicitly for this field.

Reply:
does it provide that on CharField? Any reason we are using TextField and then restricting it?

Reply:
Thinking if having a migration is better here. Having this in code and then everyone getting confused why this is a text field with length limitation in future, without the context of this change, wont be a great thing to have.
oh, we could have some records with more than 255 characters for description and migration would mess those up?

fyi @ajeety4 curious about your thoughts on this?

Reply:
@mkangia I was thinking in terms of keeping this code migration free, since that was possible. 

The TextField can be turned into a CharField pretty easily for `CaseProperty` and `CasePropertyGroup` model. I will make the migration to CharField, as that will not require adding custom validation of any sort for this `max_length` check. 

I will update you if i find any issues regarding the migration, i have not yet looked into how having a length longer than 255 on TextField will affect the migration.

Reply:
Cool. Good to give that all a thought/try, if it isn't straight forward enough and puts system at risk, I am fine with how it is now. Using pre-defined django validators like the one ajay mentioned, still sounds a good change.

Reply:
Fixed. [8a81cb5](https://github.com/dimagi/commcare-hq/pull/32878/commits/8a81cb5376dac6840aa38073b18e0aea27e57e0b)

Reply:
offlined that we would stick with current approach and use textField only, to avoid user's added description getting trimmed in migration.